test/riak_kv_sweeper_SUITE.erl

```diff
@@ -0,0 +1,892 @@
+        0 ->
+            NumKeys * WaitTimeMsecs;
+        1 ->
+            (NumKeys div 2)  * WaitTimeMsecs;
+        _ ->
+            (NumKeys div (ThrottleAfterBytes div ObjSizeBytes))  * WaitTimeMsecs
+    end.
+
+riak_object_bin(B, K, ObjSizeBytes) ->
+    riak_object:to_binary(v1, riak_object:new(B, K, <<0:ObjSizeBytes/unit:8>>)).
+
+add_hours(Hour, Inc)  ->
+    (Hour + Inc) rem 24.
+
+%% Waiting for 2500 msecs because at least 1 second must elapse
+%% before a sweep is re-run using the run_interval (now - ts) > 1,
+%% see kv_sweeper:expired
+min_scheduler_response_time_msecs() ->
+    2000.
+
+pick(List) when length(List) > 0 ->
+    N = random:uniform(length(List)),
+    lists:nth(N, List).
+
+sweep_all_indices(Indices) ->
+    sweep_all_indices(Indices, ok).
+
+sweep_all_indices(Indices, ExpectedResult) ->
+    sweep_all_indices(Indices, ExpectedResult, sweeper_callback_1).
+
+sweep_all_indices(Indices, ExpectedResult, SweepParticipant) ->
+    riak_kv_sweeper:enable_sweep_scheduling(),
+    tick_per_index(Indices),
+    ReceiveFun = fun(I) ->
+                         ExpectedResult = receive_msg({ok, successful_sweep, SweepParticipant, I})
+                 end,
+    lists:foreach(ReceiveFun, Indices).
+
+tick_per_index(Indices) ->
+    application:set_env(riak_kv, sweep_concurrency, length(Indices)),
+    %% Only one sweep scheduled per tick, so send a tick for each index:
+    lists:foreach(fun(_) -> riak_kv_sweeper:sweep_tick() end, Indices).
+
+assert_all_indices_idle(Indices) ->
+    {_, Sweeps} = riak_kv_sweeper:status(),
+    SweepData = [{riak_kv_sweeper_state:sweep_state(S),
+                  riak_kv_sweeper_state:sweep_index(S)} ||
+                    S <- Sweeps],
+    SweepIndices = lists:sort([I || {idle, I} <- SweepData]),
+    SweepIndices = Indices.
+
+receive_msg(Msg) ->
+    ok = match_retry(
+           fun() ->
+                   receive Msg0 when Msg0 == Msg ->
+                           ok
+                   after 0 ->
+                           not_ok
+                   end
+           end, ok).
+
+get_sweep_on_index(Index) ->
+    {_, Sweeps} = riak_kv_sweeper:status(),
+    IndexedSweeps = [{riak_kv_sweeper_state:sweep_index(S), S} || S <- Sweeps],
+    {Index, Sweep} = lists:keyfind(Index, 1, IndexedSweeps),
+    Sweep.
+
+is_testcase({F, 1}) ->
+    match =:= re:run(atom_to_list(F), "^test_", [{capture, none}]);
+is_testcase({_, _}) ->
+    false.
+
+match_retry(Fun, Result) ->
+    %% Put Result in a tuple to allow result to be max_retries
+    %% match_retry(Fun, max_retries)
+    match_retry(_Max = 100, _N = 1, _Sleep = 100, Fun, {'_', Result}).
+
+match_retry(Max, N, _Sleep, _Fun, {'_', _Result}) when N > Max ->
+    max_retries;
+
+match_retry(Max, N, Sleep, Fun, {'_', Result}) ->
+    case Fun() of
+        Result ->
+            Result;
+        _ ->
+            timer:sleep(trunc(Sleep)),
+            match_retry(Max, N+1, max(1000, Sleep*1.1), Fun, {'_', Result})
```

Comment:
Sure, I get that, but is it a problem to just keep checking very quickly? The code you wrote starts with a 100 ms delay, then backs off from there. Why not just keep checking every 100 ms until it succeeds? It would chew a little more CPU, but if it makes the test finish faster and simplifies our test code, that seems like it might be a worthwhile tradeoff.